woocommerce/woocommerce
# Add/api test coupons

### All Submissions:


### Changes proposed in this Pull Request:



Closes https://github.com/woocommerce/woocommerce/issues/30422 and https://github.com/woocommerce/woocommerce/issues/30678.

### How to test the changes in this Pull Request:

1. After cloning this branch, setup the e2e environment in your local.
2. Setup the `.env` file using the following values:
    ```
    BASE_URL="http://localhost:8084"
    USER_KEY="admin"
    USER_SECRET="password"
    ```
3. Run the tests for the Coupons API in this PR: `npm test -- --group=coupons`
4. All coupons tests should pass:
    <img width="379" alt="Screen Shot 2021-10-19 at 2 21 48 PM" src="https://user-images.githubusercontent.com/4509348/137854671-97074b6a-4b3a-4e7f-a06a-f147a3c3d587.png">
5. Login to WP Admin and navigate to WooCommerce > Orders. Verify that there are no orders in the list. This means that the tests correctly cleaned up the order it created.
6. Go to Marketing > Coupons. Verify that there are no coupons listed. This means that the test had correctly cleaned up all the test coupons it used.

### Changelog entry

> Added API tests for the Coupons API.

### FOR PR REVIEWER ONLY:
